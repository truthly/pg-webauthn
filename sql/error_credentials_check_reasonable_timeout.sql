BEGIN;

CREATE EXTENSION webauthn CASCADE;

SELECT jsonb_pretty(webauthn.init_credential(
  challenge := '\xf1f49abe5e3dcff7a1f522252f4fb574df415dd087aae156114ac9b51fbf4129'::bytea,
  relying_party_name := 'Localhost'::text,
  relying_party_id := 'localhost'::text,
  user_name := 'test'::text,
  user_id := '\xb3368c7317791c5a98b81428cdf3e35012aa71e6090d04930b390049ead7c282064ee24e9dc7219b6d727cc85aad4dcc0f3134f8e62c6c896a48ac08aac3db1b'::bytea,
  user_display_name := 'test'::text,
  timeout := '29 seconds'::interval,
  challenge_at := '2020-12-13 16:35:00+01'
));

ROLLBACK;
