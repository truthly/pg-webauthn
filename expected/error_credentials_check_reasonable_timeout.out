BEGIN;
CREATE EXTENSION webauthn CASCADE;
NOTICE:  installing required extension "pguecc"
NOTICE:  installing required extension "pgcrypto"
SELECT jsonb_pretty(webauthn.init_credential(
  challenge := '\xf1f49abe5e3dcff7a1f522252f4fb574df415dd087aae156114ac9b51fbf4129'::bytea,
  relying_party_name := 'Localhost'::text,
  relying_party_id := 'localhost'::text,
  user_name := 'test'::text,
  user_id := '\xb3368c7317791c5a98b81428cdf3e35012aa71e6090d04930b390049ead7c282064ee24e9dc7219b6d727cc85aad4dcc0f3134f8e62c6c896a48ac08aac3db1b'::bytea,
  user_display_name := 'test'::text,
  timeout := '29 seconds'::interval,
  challenge_at := '2020-12-13 16:35:00+01'
));
ERROR:  new row for relation "credential_challenges" violates check constraint "reasonable_timeout"
DETAIL:  Failing row contains (\xf1f49abe5e3dcff7a1f522252f4fb574df415dd087aae156114ac9b51fbf41..., Localhost, localhost, test, \xb3368c7317791c5a98b81428cdf3e35012aa71e6090d04930b390049ead7c2..., test, @ 29 secs, preferred, null, null, null, Sun Dec 13 07:35:00 2020 PST).
CONTEXT:  SQL function "init_credential" statement 1
ROLLBACK;
